Core.UnitTests/Serialization/SerializationEventRaiserTest.cs

```diff
@@ -78,20 +78,13 @@ public void InvokeAttributedMethod_UsesCache ()
       ClassWithDeserializationEvents instance = new ClassWithDeserializationEvents ();
       StreamingContext context = new StreamingContext();
 
-      MockRepository repository = new MockRepository();
-      SerializationEventRaiser eventRaiserMock = repository.StrictMock<SerializationEventRaiser>();
+      var eventRaiserMock = new Mock<SerializationEventRaiser> {CallBase = true};
 
-      eventRaiserMock.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
-      LastCall.CallOriginalMethod (OriginalCallOptions.CreateExpectation);
+      eventRaiserMock.Object.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
+      eventRaiserMock.Object.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
```

Comment:
should we move the first set of verify up here in between the two calls? Then do another one after the second call?

Reply:
My thoughts were that this way we preserve the AAA pattern and keep the test minimal without losing any faith in the test. Do you think adding the extra verification would improve your trust in the test?

Reply:
yes, it would make this more expressive. This test consists of two phases, the first one populates the cache and the second one uses the cache. If you just do assertions at the end, you don't know that the cache is working, just that the call isn't happening twice. When I write a test that has the sole purpose of documenting that the cache works, I like it to be focused on this.